gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/plan/use_case/CreatePlanUseCase.java

```diff
@@ -41,6 +41,10 @@ public Output execute(Input input) {
             throw new PlanInvalidException("Can't manually create Federated Plan");
         }
 
+        if (api.getDefinitionVersion() == DefinitionVersion.V4 && api.getApiDefinitionV4().isTcpProxy()) {
+            throw new PlanInvalidException("Cannot create mTLS plan for TCP API");
+        }
```

Comment:
I think either your test or error message is wrong here
You are just checking the version of the API and if it is a TCP proxy, but there is no test on the plan type.
So why the message is talking about mTLS plan ?

Reply:
Ok, fixed